core/src/docs/rfcs/2602_object_versioning.md

```diff
@@ -0,0 +1,78 @@
+Proposal Name: object versioning
+Start Date: 2023-07-06
+RFC PR: https://github.com/apache/incubator-opendal/pull/2602
+Tracking Issue: (leave this empty)
+
+# Summary
+
+This proposal describes the object versioning (or object version control) feature of OpenDAL.
+
+# Motivation
+
+Object versioning is a common feature in many storage services. 
+
+With object versioning, users can:
+
+- Track the history of an object.
+- Implement optimistic concurrency control.
+- Implement a simple backup system.
+
+# Guide-level explanation
```

Comment:
This section needs to be more detailed and user-friendly. We should provide clear instructions on how our users can utilize object versioning and highlight the benefits they will gain from it.

Maybe we can start with an example in AWS S3 with object version enabled.

Reply:
Nice idea.